src/utils/blockchain/unwrap.ts

```diff
@@ -0,0 +1,49 @@
+import { Web3Interface } from "../../scripts/web3-interface";
+import { archLogger } from "../../logger/chalk-theme";
+import { fetchAndDecryptShard } from "../arweave";
+import { handleRpcError } from "../rpc-error-handler";
+import { inMemoryStore } from "../onchain-data";
+
+const MAX_RETRIES = 5;
+const INTERVAL_BETWEEN_RETRIES = 5000;
+
+const wait = (ms) => new Promise((res) => setTimeout(res, ms));
+
+const unwrapSarcophagusWithRetry = async (unwrapFn: Function, depth = 0) => {
+  try {
+    return await unwrapFn();
+  } catch (e) {
+    archLogger.warn(`Unwrap attempt ${depth + 1} failed, retrying....`);
+    if (depth > MAX_RETRIES) {
+      throw e;
+    }
+
+    await wait(INTERVAL_BETWEEN_RETRIES);
+
+    return unwrapSarcophagusWithRetry(unwrapFn, depth + 1);
```

Comment:
I like loops better for readability but that's also a style thing and up to you

Reply:
I left as-is, just b/c i prefer loops when its best-case is the whole loop runs, whereas happy path here is it just runs once. but agree: readability is slightly degraded.